feat/courses-page

0123meir opened the pull request:
courses API
courses and lectures sidebar
react routes
new client route: /course-management

**course management - the person that will work on the lecture upload should use this page as a base to add lectures.**

0123meir committed:
Merge remote-tracking branch 'origin/upload-mp3-to-server' into 42-courses

0123meir committed:
basic side bar design

0123meir committed:
navigation and course creation

0123meir committed:
x

x

0123meir committed:
courses and lecture effected by onClick

0123meir committed:
Merge remote-tracking branch 'origin/create-permission-service' into 42-courses

0123meir committed:
added client and back courses design and logic

WIP, some features don't work or were not tested yet

0123meir committed:
courses e2e

Adds active lecture state management to course UI

Introduces tracking and highlighting of the active lecture in the course interface for improved navigation and user feedback. Updates state store and components to support active lecture selection and visual indication.

0123meir committed:
converted to sidebar

0123meir committed:
uncommited cahnge

0123meir committed:
Merge remote-tracking branch 'origin/Develop' into 42-courses

0123meir committed:
added api to gw

0123meir committed:
added jwt to work localy

0123meir committed:
small fixes

0123meir committed:
reverted changes to make sidebar work

0123meir committed:
Merge remote-tracking branch 'origin/42-courses' into 42-courses

0123meir committed:
remove optimistic watched lecture press

0123meir committed:
fixed some issues with the course manager

the ux is ready, the requests to the backend are still not implemented

0123meir committed:
merged the courses state and hook

lirili larila

0123meir committed:
fixed delete and add functionality to courses manager

shimpanzini bananini

0123meir committed:
fixed text overflow issue

0123meir committed:
Merge remote-tracking branch 'origin/Develop' into 42-courses

0123meir committed:
rm swagger and readme

AsafL910 commented on courses-service/src/main.ts:
```diff
@@ -0,0 +1,16 @@
+import { NestFactory } from '@nestjs/core';
+import { AppModule } from './app.module';
+import { ValidationPipe } from '@nestjs/common';
+// import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger';
```
no swagger?

0123meir replied:
1. there is some dependency problem that made me spend too much time solving - not worth it
2. how many times did you actually use swagger to read the docs of the API and didn't just open the controller?

AsafL910 replied:
On a relatively larger API like yours I think it is worth it. I think there are examples for it in other services we created but If not don't bother adding it now

AsafL910 commented on frontend/src/components/Courses/manager/AddCourseForm.tsx:
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
```
remove import react lines

AsafL910 commented on frontend/src/App.tsx:
```diff
@@ -1,19 +1,36 @@
+  }, []);
+
+  return (
+    <div>
```
no need to wrap in div

AsafL910 commented on frontend/src/components/Courses/CourseList.tsx:
```diff
@@ -0,0 +1,93 @@
+  }
+
+  if (error) {
+    return (
```
i personally prefer to return one jsx code at the bottom with conditional parameters that set what is rendered

0123meir replied:
[I like the guard-clause method better](https://deviq.com/design-patterns/guard-clause)

AsafL910 commented on frontend/src/components/Courses/manager/CourseCard.tsx:
```diff
@@ -0,0 +1,63 @@
+  onDeleteLecture: (courseIndex: number, lectureIndex: number) => void;
+}
+
+const CourseCard: React.FC<CourseCardProps> = ({
```
I prefer (props: CourseCardProps) with props.course so that you don't need to set everything at the top and easily see that each state comes from the props and not a regular state

0123meir replied:
I always do {a,b,c} :propType instead of props: propType
the props.something just adds more overhead to call the props.

AsafL910 commented on frontend/src/components/Courses/sidebar/CourseItem.tsx:
```diff
@@ -0,0 +1,106 @@
+import { FaCircle } from 'react-icons/fa';
+import { IoChevronDownOutline, IoChevronUpOutline } from 'react-icons/io5';
+
+import { detectTextDirection } from '../../../utils/detectTextDirection';
```
absolute path

AsafL910 commented on frontend/src/components/LeftSideMenu.tsx:
```diff
@@ -0,0 +1,38 @@
+        }}
+      >
+        <div className="flex items-center justify-center h-full text-gray-600">
+          {isOpen ? <FiChevronLeft size={20} /> : <FiChevronRight size={20} />}
```
use rem or at least declare the units

AsafL910 commented on frontend/src/config/config.ts:
```diff
@@ -0,0 +1,4 @@
+export const config = {
+     API_URL: import.meta.env.VITE_API_URL || 'http://localhost:3000',
+     MOCK_JWT: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjY4MGY2MGQ1MDRmZmY4NmY1NmM4M2VkZSIsInVzZXJuYW1lIjoibWVpciIsInJvbGUiOiJzdHVkZW50IiwiaWF0IjoxNzQ2MDkzNzY4LCJleHAiOjE3NDYwOTczNjh9.5IuJDK88LqYKqfLEQjs-6DA8XNGah5y95_lmQoptL8M', // Remove this when implementing real auth
```
.env?

0123meir replied:
this is a config file for easier dev usage.
there is an .env as you can see: `import.meta.env.VITE_API_URL`

AsafL910 commented on frontend/src/hooks/useApi.ts:
```diff
@@ -0,0 +1,89 @@
+    async <T>(endpoint: string, options?: RequestInit): Promise<T> => {
+      const headers = {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${config.MOCK_JWT}`, //TODO: replace with real JWT token
```
no need to declare twice, call it AUTH_TOKEN or something and get from .env

0123meir replied:
this will be obsolete the moment we have a user store

AsafL910 commented on frontend/src/hooks/useApi.ts:
```diff
@@ -0,0 +1,89 @@
+
+        return await response.json();
+      } catch (error) {
+        // Instead of throwing a new object, return a rejected promise
```
remove comment

AsafL910 commented on frontend/src/hooks/useApi.ts:
```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
remove

0123meir replied:
this is actually useful in the page.
the errors contain status attribute but TS doesn't recognize it.
it's either ignoring the type problem, which is what I did;
or, creating new type and class in order to use httpError.

I added a new type but not a class. only thing I wanted is to add a status property to the error

AsafL910 replied:
I don't think supressing these errors is a good idea unless it requires a serious amount of work to resolve. If that's the case keep it as is.

0123meir replied:
removed it already

AsafL910 commented on frontend/src/pages/CoursesMangerPage.tsx:
```diff
@@ -0,0 +1,134 @@
+    if (currentCourse.trim() && isAdmin) {
+      await addCourse({
+        name: currentCourse.trim(),
+        lecturer: [ mockLecturerId], // adjust as needed
```
Don't do it now since it will conflict with all of us but we need to remember to run prettier

0123meir replied:
When the user store will be made, I will remove this

AsafL910 replied:
I talked about prettier though

0123meir replied:
ok. I made changes in the file anyway

AsafL910 commented on frontend/src/pages/CoursesMangerPage.tsx:
```diff
@@ -0,0 +1,134 @@
+    if (newTitle !== null) {
+      const updatedCourses = [...courses];
+      updatedCourses[courseIndex].lectures[lectureIndex].title = newTitle;
+      setCourses(updatedCourses);
```
I don't think you need to deep copy here since zustand takes care of setting the courses correctly

AsafL910 commented on frontend/src/pages/CoursesMangerPage.tsx:
```diff
@@ -0,0 +1,134 @@
+    courseIndex: number,
+    lectureIndex: number
+  ): void => {
+    if (window.confirm('Are you sure you want to delete this lecture?')) {
```
maybe add TODO to create a real modal in the app

AsafL910 commented on frontend/src/state/CoursesStore.tsx:
```diff
@@ -0,0 +1,188 @@
+        lectureId,
+      });
+    } catch (err) {
+      // Rollback on error
```
remove

0123meir replied:
why, it's a comment that explains what is going on here

AsafL910 replied:
Usually a comment means the code itself is too complex. You can use function names to explain what logic is happening

0123meir replied:
ok, done.

AsafL910 reviewed (changes requested):
Nice work!! 
I didn't go over the auth service here since it was done by Liraz

0123meir committed:
rm swagger

0123meir committed:
pr

0123meir committed:
added status prop in lecture

0123meir committed:
rm unused

0123meir committed:
merged add-permissions into the branch

shaynisanov commented on courses-service/src/dto/lecture.dto.ts:
```diff
@@ -0,0 +1,56 @@
+
+  @IsOptional()
+  @IsString()
+  videoUrl?: string;
```
u dont need this videoUrl because u have requests for getting the files in the s3 service

AsafL910 committed:
Added s3 token from auth service

shaynisanov commented on frontend/src/state/CoursesStore.tsx:
```diff
@@ -0,0 +1,188 @@
+        throw new Error('API not initialized. Please call initState first.');
+      }
+
+      await get().api!.post(`/courses-service/lectures`, {
```
its now a request for the video initializer, the video initializer does the request to create new lecture in courses service,
also u need to send here the mp4

shaynisanov commented on frontend/src/pages/CoursesMangerPage.tsx:
```diff
@@ -0,0 +1,133 @@
+      const courseId = courses[courseIndex]._id;
+      addLecture(courseId, {
+        title,
+        description: '',
```
where the description that the user is writing? and all the other data

0123meir replied:
when the popup will be made, a description section should be added

0123meir committed:
mv component location

0123meir committed:
pr

0123meir committed:
Merge remote-tracking branch 'origin/create-permission-service' into 42-courses

liraz581 committed:
update user handling in authentication flow to use _id and store user data in login and register pages

liraz581 committed:
changed _id in store to id and made backend send corrected data

0123meir committed:
cherry-pick from permissions branch. and small fix

0123meir committed:
changed mock user id and jwt to use actual permissions

0123meir committed:
Replaces lecturer schema with user model for courses

Removes the separate lecturer schema and instead associates courses directly with user entities, updating references and population logic accordingly. Updates course DTOs, schema, and related frontend types and state to support the new model. Adds user session restoration in the frontend using JWT and improves authentication flows. Cleans up obsolete files and ensures user uniqueness on registration.

user management and role-based access improvements and fixes

0123meir committed:
pr fixes

0123meir committed:
cleaned up the lecture object

0123meir committed:
pr

0123meir committed:
Merge branch 'Develop' into 42-courses

0123meir committed:
Merge pull request #61 from 0123meir/stream-video-from-s3

Added s3 token from auth service
